### ISSUE TITLE
Incompatible with plugins using Scala 2.13

### ISSUE THREAD

**Issue by tlazaro** [zladovan/gradle-avrohugger-plugin#8]
Hi, I wanted to use this plugin in a project with many other plugins. A few of them are built with Scala 2.13 (spotless and scalafix) while avrohugger plugin uses 2.12. During Gradle's runtime there will only be one version of Scala loaded and these just won't work together.

Would it be possible to release the plugin for 2.12 and 2.13?

**Comment by tnielens-centrica** [comment]
About this issue, best would be to run avrohugger task in a subprocess with a dedicated classpath to avoid affecting the main process classpath.

**Comment by tnielens** [comment]
About this issue, best would be to run avrohugger task in a subprocess with a dedicated classpath to avoid affecting the main process classpath.

**Comment by zladovan** [comment]
I was considering also building and publishing separate version using `com.julianpeeters:avrohugger-core_2.13` while building it. 

So there could be three variants:
 - `com.zlad.gradle.avrohugger_2.12`
 - `com.zlad.gradle.avrohugger_2.13`
 - `com.zlad.gradle.avrohugger` (alias for one of them)

But then I am not sure, if alias should keep pointing to scala 2.12, or is scala 2.13 mostly used now ?

**Comment by tlazaro** [comment]
FWIW I ended up calling avrohugger tools from the command line with a custom gradle task.

Maybe the plugin could do something like this and not depend on Scala?

```
def configAvro(project) {
    // Ideally we would use https://github.com/zladovan/gradle-avrohugger-plugin but it's built for Scala 2.12 and incompatible with other plugins

    project.configurations {
        avroJavaTools
        avroScalaTools
    }
    project.dependencies {
        avroJavaTools "org.apache.avro:avro-tools:${avroVersion}"
        avroScalaTools "com.julianpeeters:avrohugger-tools_2.12:1.3.1"
    }

    avroGenTask(project, "main", "java")
    avroGenTask(project, "test", "java")
    avroGenTask(project, "main", "scala")
    avroGenTask(project, "test", "scala")
}
```

```
avroGenTask(...) {
  ...
  if (is scala) {
    ...
    javaexec {
      main = "avrohugger.tool.Main"
      classpath = configClasspath
      args = [
        "generate-specific",
        "schema",
        input.toString(),
         output.toString()
      ]
    }
  } else {
    ...
  }
}
```

Seems like this is what @tnielens suggested?

**Comment by tnielens-centrica** [comment]
Yeah this is a common way to run jvm-based tasks for plugins. Run them with a dedicated classpath as a java subprocess.

**Comment by zladovan** [comment]
Currently there is also `typeMapping` block which allows to configure [custom type mappings](https://github.com/julianpeeters/avrohugger?tab=readme-ov-file#customizable-type-mapping) and it's values are just wrappers around `avrohugger.types` classes. This would be need to be re-implemented in some more dynamic way. Also it looks like not supported by `avrohugger-tools`.

**Comment by joan38** [comment]
I wouldn't overthink this. Just move to 2.13, there has been no breaking change in the lib since 2.13, even Scala 3 uses the Scala 2.13 lib.
People can use an older version of the plugin if they really want to stay in the past.

**Comment by zladovan** [comment]
OK. I will release it with 2.13 binaries as new 1.x version and keep 0.x for 2.12 to be able to add some patch later, if needed.

Then I have already some local work in progress with update to do not bring any scala deps, so user could pick avrohugger/scala version in build config. This could be then released as 2.x. But it can take some time to be ready.

**Comment by zladovan** [comment]
Use plugin in version 1.x when you need scala 2.13, keep using version 0.x when you need scala 2.12.

Thanks to @joan38 for contribution.
